File: mobile/app/src/main/java/com/plugplay/plugplaymobile/data/model/ProductDto.kt
package com.plugplay.plugplaymobile.data.model

import com.google.gson.annotations.SerializedName

data class ProductDto(
    @SerializedName("id") val id: Int,
    @SerializedName("name") val name: String?,
    @SerializedName("price") val price: Double?,
    @SerializedName("description") val description: String?,
    @SerializedName("stockQuantity") val stockQuantity: Int?,
    @SerializedName("pictureUrls") val pictureUrls: List<String>?,
    @SerializedName("category") val category: CategoryDto?,


    @SerializedName("reviews") val reviews: List<ReviewDto>?
)

File: backend/PlugPlay/PlugPlay.Api/Dto/ProductDto.cs
using PlugPlay.Domain.Entities;

namespace PlugPlay.Api.Dto;

public record ProductDto(
    int Id,
    string Name,
    string Description,
    decimal Price,
    int StockQuantity,
    DateTime CreatedAt,
    CategoryDto? Category,
    IEnumerable<string> PictureUrls,
    IEnumerable<ReviewDto> Reviews,
    IEnumerable<AttributeDto> Attributes);